include/revng/Pipeline/InvalidationEvent.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Pipeline/Runner.h"
+
+namespace pipeline {
+
+class InvalidationEvent {
+public:
+  llvm::Error run(Runner &Runner) const;
+  virtual void getInvalidations(const Runner &Runner,
+                                Runner::InvalidationMap &Out) const = 0;
```

Comment:
* `computeInvalidations`?
* Doc
* Why not returning it? Always in order to accumulate? Probably more efficient but it seems coupled to its users.
* Do we need a `Runner` reference here? Maybe there's a more specific object we can pass in?

Reply:
yeah computeInvalidations is better
yeah, to accumulate, calculating invalidations is usually expensive.
it needs the state of the pipeline and the context, which is pretty much a entire runner.

Reply:
* Please address all the comments above.
* `Runner::getInvalidations` is also misleading.
* Are you sure that accepting a `Runner` is the right thing? I like more `ModelInvalidationRule`'s approach which accepts `const pipeline::ContainerBase &`. Maybe the runner should have the responsibility of invoking  `getInvalidations` on all the containers?